qdl.c

```diff
@@ -464,6 +477,8 @@ int main(int argc, char **argv)
 
 	prog_mbn = argv[optind++];
 
+	xmlInitParser();
```

Comment:
This doesn't seem to fall into the same category as the other fixes in this patch; you have freeing of memory, some style changes and this.

I think it would be nice if this was 3 different patches.